src/controllers/users.ts

```diff
@@ -36,6 +36,30 @@ export async function createUser(
   }
 }
 
+export async function loginUser(
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): Promise<Response | void> {
+  try {
+    const { email, password } = req.body;
+    const result = await userService.authenticateUser(email, password);
+
+    if (!result.success) {
+      return res.status(status.UNAUTHORIZED).json({ message: result.message });
+    }
+
+    return res.status(status.OK).json({
+      message: result.message,
+      token: result.token,
+      user: result.user,
```

Comment:
Check this, Avoid exposing sensitive information such as passwords or tokens in the response payload or logs.

Reply:
<img width="244" height="415" alt="image" src="https://github.com/user-attachments/assets/9db535a4-80da-4e25-936f-b689861ea439" />

```typescript
export const userMapper = (user): IUser => ({
  name: user.name,
  age: user.age,
  likes: user.likes || DEFAULT_EMPTY,
});

```